config-provisioning/src/main/java/com/yahoo/config/provision/DockerImage.java

```diff
@@ -55,25 +76,36 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         DockerImage that = (DockerImage) o;
-        return repository.equals(that.repository) &&
-                tag.equals(that.tag);
+        return registry.equals(that.registry) &&
+               repository.equals(that.repository) &&
+               tag.equals(that.tag);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(repository, tag);
+        return Objects.hash(registry, repository, tag);
     }
 
-    public static DockerImage fromString(String name) {
-        if (name.isEmpty()) return EMPTY;
+    public static DockerImage from(String registry, String repository) {
+        return new DockerImage(registry, repository, Optional.empty());
+    }
+
+    public static DockerImage fromString(String s) {
+        if (s.isEmpty()) return EMPTY;
+
+        int firstPathSeparator = s.indexOf('/');
+        if (firstPathSeparator < 0) throw new IllegalArgumentException("Missing path separator in '" + s + "'");
```

Comment:
I think we should allow specifying docker image without registry

Reply:
Explicit is better I think. Won't the default depend on the container engine? I believe Docker uses hub.docker.com without registry, but that might not be the case for e.g. Podman?